apps/dashboard/src/lib/components/LMS/SideBar.svelte

```diff
@@ -87,19 +95,21 @@
 
       <ul class="my-5">
         {#each sideLinks as item (item.name)}
-          <a href={item.link} class="text-black" on:click={toggleSidebar}>
-            <li
-              class="flex items-center py-3 px-4 mb-2 {NavClasses.item} {isActive(
-                $page.url.pathname,
-                `${item.link}`
-              )
-                ? NavClasses.active
-                : 'dark:text-white'}"
-            >
-              <svelte:component this={item.icon} size={24} class="carbon-icon dark:fill-[#fff]" />
-              <p class="dark:text-white ml-2">{item.name}</p>
-            </li>
-          </a>
+          {#if !item.show || (typeof item.show === 'function' && item.show())}
```

Comment:
To avoid this custom if statement logic here, you can already filter out the sideLinks in the reactive block before it is even rendered.

Something like:

```
$: sideLinks = [
  // all links here
].filter((link) => (link.show ? link.show() : true));
```